src/sentry/models/dsymfile.py

```diff
@@ -12,23 +12,221 @@
 import shutil
 import hashlib
 import tempfile
-from django.db import models, transaction, IntegrityError
+from django.db import models, transaction, connection, IntegrityError
 
 try:
-    from symsynd.mach import get_macho_uuids
+    from symsynd.macho.arch import get_macho_uuids
     have_symsynd = True
 except ImportError:
     have_symsynd = False
 
-from sentry.db.models import FlexibleForeignKey, Model, sane_repr
+from sentry.db.models import FlexibleForeignKey, Model, BoundedBigIntegerField, \
+    sane_repr, BaseManager
 from sentry.models.file import File
 from sentry.utils.zip import safe_extract_zip
 
 
+MAX_SYM = 256
 KNOWN_DSYM_TYPES = {
     'application/x-mach-binary': 'macho'
 }
 
+SDK_MAPPING = {
+    'iPhone OS': 'iOS',
+    'tvOS': 'tvOS',
+    'Mac OS': 'macOS',
+}
+
+
+def get_sdk_from_system_info(info):
+    if not info:
+        return None
+    try:
+        sdk_name = SDK_MAPPING[info['system_name']]
+        system_version = tuple(int(x) for x in (
+            info['system_version'] + '.0' * 3).split('.')[:3])
+    except LookupError:
+        return None
+
+    return {
+        'dsym_type': 'macho',
+        'sdk_name': sdk_name,
+        'version_major': system_version[0],
+        'version_minor': system_version[1],
+        'version_patchlevel': system_version[2],
+    }
+
+
+class DSymSDKManager(BaseManager):
+
+    def enumerate_sdks(self, sdk=None, version=None):
+        """Return a grouped list of SDKs."""
+        filter = ''
+        args = []
+        if version is not None:
+            for col, val in zip(['major', 'minor', 'patchlevel'],
+                                version.split('.')):
+                if not val.isdigit():
+                    return []
+                filter += ' and k.version_%s = %d' % (
+                    col,
+                    int(val)
+                )
+        if sdk is not None:
+            filter += ' and k.sdk_name = %s'
+            args.append(sdk)
+        cur = connection.cursor()
+        cur.execute('''
+   select distinct k.*, count(b.*) as bundle_count, o.cpu_name
+              from sentry_dsymsdk k,
+                   sentry_dsymbundle b,
+                   sentry_dsymobject o
+             where b.sdk_id = k.id and
+                   b.object_id = o.id %s
+          group by k.id, k.sdk_name, o.cpu_name
+        ''' % filter, args)
```

Comment:
This seems like something now that can be converted into ORM syntax without joining strings and piecing together this SQL.